File: src/ui/api/routers/job_list.py
"""Job List API Router"""

from fastapi import APIRouter, Depends, HTTPException, Query, BackgroundTasks
from typing import Optional, List
import logging

from ..dependencies import get_rag
from ..services.job_list_service import JobListService, get_job_list_service
from ..models.job_list_models import (
    JobSearchRequest,
    JobSearchResponse,
    JobListing,
    JobListingBrief,
    Application,
    ApplicationCreate,
    ApplicationUpdate,
    ApplicationListResponse,
    SavedSearch,
    SavedSearchCreate,
    JobRecommendation,
    CoverLetterRequest,
    CoverLetterResponse,
    JobSearchStats,
    ApplicationStats,
    ApplicationStatus,
    ScrapeStatus,
)
from ..models.responses import ErrorResponse
from ..scrapers.google_dorking_scraper import get_dork_strategies, get_dork_categories

logger = logging.getLogger(__name__)

router = APIRouter(prefix="/api/job-list", tags=["job-list"])

# Service dependency
_service: Optional[JobListService] = None


def get_service() -> JobListService:
    """Get or create JobListService instance"""
    global _service
    if _service is None:
        rag = get_rag()
        _service = get_job_list_service(rag)
    return _service


# ============== Search Endpoints ==============

@router.post(
    "/search",
    response_model=JobSearchResponse,
    responses={400: {"model": ErrorResponse}, 500: {"model": ErrorResponse}},
    summary="Search for jobs",
    description="Search for jobs using natural language query and/or structured filters",
)
async def search_jobs(
    request: JobSearchRequest,
    service: JobListService = Depends(get_service),
) -> JobSearchResponse:
    """
    Search for jobs with natural language and filters.

    Supports:
    - Natural language queries: "remote ML engineer jobs $150k+ at startups"
    - Structured filters: location, salary, company size, sources
    - Resume matching: returns match scores against your resume
    - Pagination and sorting
    """
    try:
        if not request.query and not request.filters:
            raise HTTPException(
                status_code=400,
                detail="Either query or filters must be provided"
            )
        return await service.search_jobs(request)
    except Exception as e:
        logger.error(f"Search error: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.get(
    "/jobs",
    response_model=JobSearchResponse,
    summary="Get cached jobs",
    description="Get paginated list of cached jobs with optional filters",
)
async def get_jobs(
    page: int = Query(1, ge=1),
    limit: int = Query(20, ge=1, le=100),
    sort_by: str = Query("posted_date", pattern="^(match_score|posted_date|salary)$"),
    sort_order: str = Query("desc", pattern="^(asc|desc)$"),
    location_type: Optional[str] = Query(None, description="Filter by remote/hybrid/onsite"),
    source: Optional[str] = Query(None, description="Filter by source"),
    service: JobListService = Depends(get_service),
) -> JobSearchResponse:
    """Get paginated jobs from cache"""
    from ..models.job_list_models import JobFilters, LocationType, JobSource

    filters = JobFilters()
    if location_type:
        filters.location_type = [LocationType(location_type)]
    if source:
        filters.sources = [JobSource(source)]

    request = JobSearchRequest(
        filters=filters,
        page=page,
        limit=limit,
        sort_by=sort_by,
        sort_order=sort_order,
        include_match_scores=True,
    )

    return await service.search_jobs(request)


@router.get(
    "/jobs/{job_id}",
    response_model=JobListing,
    responses={404: {"model": ErrorResponse}},
    summary="Get job details",
    description="Get full job details with match analysis",
)
async def get_job_details(
    job_id: str,
    service: JobListService = Depends(get_service),
) -> JobListing:
    """Get detailed job information with match scores"""
    job = await service.get_job_details(job_id)
    if not job:
        raise HTTPException(status_code=404, detail="Job not found")
    return job


# ============== Recommendations ==============

@router.get(
    "/recommendations",
    response_model=List[JobRecommendation],
    summary="Get AI job recommendations",
    description="Get personalized job recommendations based on your resume",
)
async def get_recommendations(
    limit: int = Query(10, ge=1, le=50),
    service: JobListService = Depends(get_service),
) -> List[JobRecommendation]:
    """Get AI-powered job recommendations"""
    return await service.get_recommendations(limit)


# ============== Application Tracking ==============

@router.get(
    "/applications",
    response_model=ApplicationListResponse,
    summary="Get tracked applications",
    description="Get all tracked job applications with status",
)
async def get_applications(
    status: Optional[ApplicationStatus] = None,
    limit: int = Query(50, ge=1, le=200),
    offset: int = Query(0, ge=0),
    service: JobListService = Depends(get_service),
) -> ApplicationListResponse:
    """Get tracked applications"""
    return service.get_applications(status=status, limit=limit, offset=offset)


@router.post(
    "/applications",
    response_model=Application,
    responses={400: {"model": ErrorResponse}},
    summary="Track a job application",
    description="Start tracking a job application",
)
async def create_application(
    data: ApplicationCreate,
    service: JobListService = Depends(get_service),
) -> Application:
    """Create or update application tracking"""
    try:
        return service.create_application(data)
    except Exception as e:
        logger.error(f"Create application error: {e}")
        raise HTTPException(status_code=400, detail=str(e))


@router.get(
    "/applications/{app_id}",
    response_model=Application,
    responses={404: {"model": ErrorResponse}},
    summary="Get application details",
)
async def get_application(
    app_id: str,
    service: JobListService = Depends(get_service),
) -> Application:
    """Get application details with timeline"""
    app = service.get_application(app_id)
    if not app:
        raise HTTPException(status_code=404, detail="Application not found")
    return app


@router.put(
    "/applications/{app_id}",
    response_model=Application,
    responses={404: {"model": ErrorResponse}},
    summary="Update application status",
)
async def update_application(
    app_id: str,
    data: ApplicationUpdate,
    service: JobListService = Depends(get_service),
) -> Application:
    """Update application status or details"""
    app = service.update_application(app_id, data)
    if not app:
        raise HTTPException(status_code=404, detail="Application not found")
    return app


@router.delete(
    "/applications/{app_id}",
    summary="Delete application tracking",
)
async def delete_application(
    app_id: str,
    service: JobListService = Depends(get_service),
) -> dict:
    """Stop tracking an application"""
    success = service.delete_application(app_id)
    if not success:
        raise HTTPException(status_code=404, detail="Application not found")
    return {"success": True, "message": "Application deleted"}


@router.get(
    "/applications/reminders/due",
    response_model=List[Application],
    summary="Get due reminders",
)
async def get_due_reminders(
    service: JobListService = Depends(get_service),
) -> List[Application]:
    """Get applications with reminders due today or overdue"""
    return service.get_due_reminders()


# ============== Cover Letter Generation ==============

@router.post(
    "/jobs/{job_id}/cover-letter",
    response_model=CoverLetterResponse,
    responses={404: {"model": ErrorResponse}},
    summary="Generate cover letter",
    description="Generate a tailored cover letter for a job",
)
async def generate_cover_letter(
    job_id: str,
    request: Optional[CoverLetterRequest] = None,
    service: JobListService = Depends(get_service),
) -> CoverLetterResponse:
    """Generate AI cover letter for a job"""
    try:
        custom_prompt = request.custom_prompt if request else None
        cover_letter = await service.generate_cover_letter(job_id, custom_prompt)

        # Get job for response
        job = await service.get_job_details(job_id)

        return CoverLetterResponse(
            job_id=job_id,
            cover_letter=cover_letter,
            word_count=len(cover_letter.split()),
            highlights_used=job.matched_skills if job else [],
        )
    except ValueError as e:
        raise HTTPException(status_code=404, detail=str(e))
    except Exception as e:
        logger.error(f"Cover letter generation error: {e}")
        raise HTTPException(status_code=500, detail=str(e))


# ============== Saved Searches ==============

@router.get(
    "/saved-searches",
    response_model=List[SavedSearch],
    summary="Get saved searches",
)
async def get_saved_searches(
    service: JobListService = Depends(get_service),
) -> List[SavedSearch]:
    """Get all saved search presets"""
    return service.get_saved_searches()


@router.post(
    "/saved-searches",
    response_model=SavedSearch,
    summary="Save a search",
)
async def save_search(
    data: SavedSearchCreate,
    service: JobListService = Depends(get_service),
) -> SavedSearch:
    """Save a search preset for quick access"""
    return service.save_search(data)


@router.delete(
    "/saved-searches/{search_id}",
    summary="Delete saved search",
)
async def delete_saved_search(
    search_id: str,
    service: JobListService = Depends(get_service),
) -> dict:
    """Delete a saved search"""
    success = service.delete_saved_search(search_id)
    if not success:
        raise HTTPException(status_code=404, detail="Search not found")
    return {"success": True, "message": "Search deleted"}


# ============== Scraping Control ==============

@router.post(
    "/scrape",
    summary="Trigger job scraping",
    description="Manually trigger scraping from job sites",
)
async def trigger_scrape(
    keywords: List[str] = Query(..., min_length=1),
    location: Optional[str] = None,
    sources: Optional[List[str]] = None,
    background_tasks: BackgroundTasks = None,
    service: JobListService = Depends(get_service),
) -> dict:
    """Trigger background scraping"""
    task_id = await service.trigger_scrape(keywords, location, sources)
    return {
        "task_id": task_id,
        "status": "started",
        "message": f"Scraping started for keywords: {keywords}"
    }


@router.get(
    "/scrape/status/{task_id}",
    summary="Get scrape status",
)
async def get_scrape_status(
    task_id: str,
    service: JobListService = Depends(get_service),
) -> dict:
    """Get status of a scraping task"""
    status = service.get_scrape_status(task_id)
    return {"task_id": task_id, "status": status.value}


# ============== Google Dorking ==============

@router.get(
    "/dork-strategies",
    summary="Get available dork strategies",
    description="Get all available Google dorking strategies organized by category",
)
async def get_dork_strategies_endpoint() -> dict:
    """Get all dork strategies for the UI selector"""
    return get_dork_strategies()


@router.get(
    "/dork-categories",
    summary="Get dork categories",
    description="Get list of dork categories for dropdown",
)
async def get_dork_categories_endpoint() -> list:
    """Get dork categories for dropdown"""
    return get_dork_categories()


# ============== Statistics ==============

@router.get(
    "/stats/jobs",
    response_model=JobSearchStats,
    summary="Get job statistics",
)
async def get_job_stats(
    service: JobListService = Depends(get_service),
) -> JobSearchStats:
    """Get job database statistics"""
    stats = service.get_job_stats()
    return JobSearchStats(
        total_jobs_indexed=stats.get("total_jobs", 0),
        jobs_by_source=stats.get("by_source", {}),
        jobs_by_location_type=stats.get("by_location_type", {}),
        average_salary=stats.get("average_salary"),
        last_scrape_at=stats.get("last_scrape"),
    )


@router.get(
    "/stats/applications",
    response_model=ApplicationStats,
    summary="Get application statistics",
)
async def get_application_stats(
    service: JobListService = Depends(get_service),
) -> ApplicationStats:
    """Get application tracking statistics"""
    stats = service.get_application_stats()
    return ApplicationStats(
        total_applications=stats.get("total", 0),
        by_status=stats.get("by_status", {}),
        response_rate=stats.get("response_rate"),
    )

File: backend-go/internal/api/handlers/joblist.go
package handlers

import (
	"context"

	"github.com/gofiber/fiber/v2"
	"github.com/google/uuid"

	"github.com/resume-rag/backend/internal/domain"
)

// JobListService defines the interface for job list operations
type JobListService interface {
	Search(ctx context.Context, req domain.JobSearchRequest) (*domain.JobSearchResponse, error)
	GetJobs(ctx context.Context, page, limit int, sortBy, sortOrder string, filters *domain.JobFilters) (*domain.JobSearchResponse, error)
	GetJobDetails(ctx context.Context, jobID uuid.UUID) (*domain.Job, error)
	GetRecommendations(ctx context.Context, limit int) ([]domain.JobRecommendation, error)

	// Applications
	GetApplications(ctx context.Context, status *domain.ApplicationStatus, limit, offset int) (*domain.ApplicationListResponse, error)
	CreateApplication(ctx context.Context, req domain.ApplicationCreate) (*domain.Application, error)
	GetApplication(ctx context.Context, appID uuid.UUID) (*domain.Application, error)
	UpdateApplication(ctx context.Context, appID uuid.UUID, req domain.ApplicationUpdate) (*domain.Application, error)
	DeleteApplication(ctx context.Context, appID uuid.UUID) error
	GetDueReminders(ctx context.Context) ([]domain.Application, error)

	// Cover letter
	GenerateCoverLetter(ctx context.Context, jobID uuid.UUID, customPrompt *string) (*domain.CoverLetterResponse, error)

	// Saved searches
	GetSavedSearches(ctx context.Context) ([]domain.SavedSearch, error)
	SaveSearch(ctx context.Context, req domain.SavedSearchCreate) (*domain.SavedSearch, error)
	DeleteSavedSearch(ctx context.Context, searchID uuid.UUID) error

	// Scraping
	TriggerScrape(ctx context.Context, keywords []string, location *string, sources []string) (*domain.ScrapeTask, error)
	GetScrapeStatus(ctx context.Context, taskID uuid.UUID) (*domain.ScrapeTask, error)

	// Statistics
	GetJobStats(ctx context.Context) (*domain.JobSearchStats, error)
	GetApplicationStats(ctx context.Context) (*domain.ApplicationStats, error)
}

// JobListHandler handles job list API requests
type JobListHandler struct {
	service JobListService
}

// NewJobListHandler creates a new job list handler
func NewJobListHandler(service JobListService) *JobListHandler {
	return &JobListHandler{service: service}
}

// Search handles POST /api/job-list/search
func (h *JobListHandler) Search(c *fiber.Ctx) error {
	var req domain.JobSearchRequest
	if err := c.BodyParser(&req); err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_request",
			"message": "Invalid request body",
		})
	}

	// Validate
	if req.Query == nil && req.Filters == nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_request",
			"message": "Either query or filters must be provided",
		})
	}

	// Set defaults
	if req.Page == 0 {
		req.Page = 1
	}
	if req.Limit == 0 {
		req.Limit = 20
	}
	if req.SortBy == "" {
		req.SortBy = "match_score"
	}
	if req.SortOrder == "" {
		req.SortOrder = "desc"
	}

	result, err := h.service.Search(c.Context(), req)
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "search_failed",
			"message": err.Error(),
		})
	}

	return c.JSON(result)
}

// GetJobs handles GET /api/job-list/jobs
func (h *JobListHandler) GetJobs(c *fiber.Ctx) error {
	page := c.QueryInt("page", 1)
	limit := c.QueryInt("limit", 20)
	sortBy := c.Query("sort_by", "posted_date")
	sortOrder := c.Query("sort_order", "desc")

	// Parse filters
	var filters *domain.JobFilters
	locationType := c.Query("location_type")
	source := c.Query("source")

	if locationType != "" || source != "" {
		filters = &domain.JobFilters{}
		if locationType != "" {
			filters.LocationTypes = []domain.LocationType{domain.LocationType(locationType)}
		}
		if source != "" {
			filters.Sources = []domain.JobSource{domain.JobSource(source)}
		}
	}

	result, err := h.service.GetJobs(c.Context(), page, limit, sortBy, sortOrder, filters)
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "fetch_failed",
			"message": err.Error(),
		})
	}

	return c.JSON(result)
}

// GetJobDetails handles GET /api/job-list/jobs/:job_id
func (h *JobListHandler) GetJobDetails(c *fiber.Ctx) error {
	jobID, err := uuid.Parse(c.Params("job_id"))
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_id",
			"message": "Invalid job ID format",
		})
	}

	job, err := h.service.GetJobDetails(c.Context(), jobID)
	if err != nil {
		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{
			"error":   "not_found",
			"message": "Job not found",
		})
	}

	return c.JSON(job)
}

// GetRecommendations handles GET /api/job-list/recommendations
func (h *JobListHandler) GetRecommendations(c *fiber.Ctx) error {
	limit := c.QueryInt("limit", 10)

	recommendations, err := h.service.GetRecommendations(c.Context(), limit)
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "fetch_failed",
			"message": err.Error(),
		})
	}

	return c.JSON(recommendations)
}

// GetApplications handles GET /api/job-list/applications
func (h *JobListHandler) GetApplications(c *fiber.Ctx) error {
	limit := c.QueryInt("limit", 50)
	offset := c.QueryInt("offset", 0)

	var status *domain.ApplicationStatus
	if s := c.Query("status"); s != "" {
		st := domain.ApplicationStatus(s)
		status = &st
	}

	result, err := h.service.GetApplications(c.Context(), status, limit, offset)
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "fetch_failed",
			"message": err.Error(),
		})
	}

	return c.JSON(result)
}

// CreateApplication handles POST /api/job-list/applications
func (h *JobListHandler) CreateApplication(c *fiber.Ctx) error {
	var req domain.ApplicationCreate
	if err := c.BodyParser(&req); err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_request",
			"message": "Invalid request body",
		})
	}

	app, err := h.service.CreateApplication(c.Context(), req)
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "create_failed",
			"message": err.Error(),
		})
	}

	return c.Status(fiber.StatusCreated).JSON(app)
}

// GetApplication handles GET /api/job-list/applications/:app_id
func (h *JobListHandler) GetApplication(c *fiber.Ctx) error {
	appID, err := uuid.Parse(c.Params("app_id"))
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_id",
			"message": "Invalid application ID format",
		})
	}

	app, err := h.service.GetApplication(c.Context(), appID)
	if err != nil {
		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{
			"error":   "not_found",
			"message": "Application not found",
		})
	}

	return c.JSON(app)
}

// UpdateApplication handles PUT /api/job-list/applications/:app_id
func (h *JobListHandler) UpdateApplication(c *fiber.Ctx) error {
	appID, err := uuid.Parse(c.Params("app_id"))
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_id",
			"message": "Invalid application ID format",
		})
	}

	var req domain.ApplicationUpdate
	if err := c.BodyParser(&req); err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_request",
			"message": "Invalid request body",
		})
	}

	app, err := h.service.UpdateApplication(c.Context(), appID, req)
	if err != nil {
		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{
			"error":   "not_found",
			"message": "Application not found",
		})
	}

	return c.JSON(app)
}

// DeleteApplication handles DELETE /api/job-list/applications/:app_id
func (h *JobListHandler) DeleteApplication(c *fiber.Ctx) error {
	appID, err := uuid.Parse(c.Params("app_id"))
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_id",
			"message": "Invalid application ID format",
		})
	}

	if err := h.service.DeleteApplication(c.Context(), appID); err != nil {
		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{
			"error":   "not_found",
			"message": "Application not found",
		})
	}

	return c.JSON(fiber.Map{
		"success": true,
		"message": "Application deleted",
	})
}

// GetDueReminders handles GET /api/job-list/applications/reminders/due
func (h *JobListHandler) GetDueReminders(c *fiber.Ctx) error {
	apps, err := h.service.GetDueReminders(c.Context())
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "fetch_failed",
			"message": err.Error(),
		})
	}

	return c.JSON(apps)
}

// GenerateCoverLetter handles POST /api/job-list/jobs/:job_id/cover-letter
func (h *JobListHandler) GenerateCoverLetter(c *fiber.Ctx) error {
	jobID, err := uuid.Parse(c.Params("job_id"))
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_id",
			"message": "Invalid job ID format",
		})
	}

	var req struct {
		CustomPrompt *string `json:"custom_prompt"`
	}
	_ = c.BodyParser(&req) // Optional body

	result, err := h.service.GenerateCoverLetter(c.Context(), jobID, req.CustomPrompt)
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "generation_failed",
			"message": err.Error(),
		})
	}

	return c.JSON(result)
}

// GetSavedSearches handles GET /api/job-list/saved-searches
func (h *JobListHandler) GetSavedSearches(c *fiber.Ctx) error {
	searches, err := h.service.GetSavedSearches(c.Context())
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "fetch_failed",
			"message": err.Error(),
		})
	}

	return c.JSON(searches)
}

// SaveSearch handles POST /api/job-list/saved-searches
func (h *JobListHandler) SaveSearch(c *fiber.Ctx) error {
	var req domain.SavedSearchCreate
	if err := c.BodyParser(&req); err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_request",
			"message": "Invalid request body",
		})
	}

	search, err := h.service.SaveSearch(c.Context(), req)
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "save_failed",
			"message": err.Error(),
		})
	}

	return c.Status(fiber.StatusCreated).JSON(search)
}

// DeleteSavedSearch handles DELETE /api/job-list/saved-searches/:search_id
func (h *JobListHandler) DeleteSavedSearch(c *fiber.Ctx) error {
	searchID, err := uuid.Parse(c.Params("search_id"))
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_id",
			"message": "Invalid search ID format",
		})
	}

	if err := h.service.DeleteSavedSearch(c.Context(), searchID); err != nil {
		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{
			"error":   "not_found",
			"message": "Search not found",
		})
	}

	return c.JSON(fiber.Map{
		"success": true,
		"message": "Search deleted",
	})
}

// TriggerScrape handles POST /api/job-list/scrape
func (h *JobListHandler) TriggerScrape(c *fiber.Ctx) error {
	var req struct {
		Keywords []string  `json:"keywords"`
		Location *string   `json:"location"`
		Sources  []string  `json:"sources"`
	}

	// Also support query params
	keywords := c.QueryArray("keywords")
	if len(keywords) == 0 {
		if err := c.BodyParser(&req); err != nil || len(req.Keywords) == 0 {
			return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
				"error":   "invalid_request",
				"message": "Keywords are required",
			})
		}
		keywords = req.Keywords
	}

	location := c.Query("location")
	if location == "" && req.Location != nil {
		location = *req.Location
	}
	var locationPtr *string
	if location != "" {
		locationPtr = &location
	}

	sources := c.QueryArray("sources")
	if len(sources) == 0 {
		sources = req.Sources
	}

	task, err := h.service.TriggerScrape(c.Context(), keywords, locationPtr, sources)
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "scrape_failed",
			"message": err.Error(),
		})
	}

	return c.Status(fiber.StatusAccepted).JSON(fiber.Map{
		"task_id": task.ID,
		"status":  task.Status,
		"message": "Scraping started",
	})
}

// GetScrapeStatus handles GET /api/job-list/scrape/status/:task_id
func (h *JobListHandler) GetScrapeStatus(c *fiber.Ctx) error {
	taskID, err := uuid.Parse(c.Params("task_id"))
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error":   "invalid_id",
			"message": "Invalid task ID format",
		})
	}

	task, err := h.service.GetScrapeStatus(c.Context(), taskID)
	if err != nil {
		return c.Status(fiber.StatusNotFound).JSON(fiber.Map{
			"error":   "not_found",
			"message": "Task not found",
		})
	}

	return c.JSON(task)
}

// GetJobStats handles GET /api/job-list/stats/jobs
func (h *JobListHandler) GetJobStats(c *fiber.Ctx) error {
	stats, err := h.service.GetJobStats(c.Context())
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "fetch_failed",
			"message": err.Error(),
		})
	}

	return c.JSON(stats)
}

// GetApplicationStats handles GET /api/job-list/stats/applications
func (h *JobListHandler) GetApplicationStats(c *fiber.Ctx) error {
	stats, err := h.service.GetApplicationStats(c.Context())
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error":   "fetch_failed",
			"message": err.Error(),
		})
	}

	return c.JSON(stats)
}
